src/main/java/io/vertx/kafka/client/producer/impl/KafkaWriteStreamImpl.java

```diff
@@ -178,6 +178,42 @@ public void end(Handler<AsyncResult<Void>> handler) {
     }
   }
 
+  @Override
+  public KafkaWriteStream<K, V> initTransactions(Handler<AsyncResult<Void>> handler) {
+    this.context.executeBlocking(future -> {
+      this.producer.initTransactions();
+      future.complete();
+    }, handler);
+    return this;
+  }
+
+  @Override
+  public KafkaWriteStream<K, V> beginTransaction(Handler<AsyncResult<Void>> handler) {
+    this.context.executeBlocking(future -> {
+      this.producer.beginTransaction();
+      future.complete();
+    }, handler);
+    return this;
+  }
+
+  @Override
+  public KafkaWriteStream<K, V> commitTransaction(Handler<AsyncResult<Void>> handler) {
+    this.context.executeBlocking(future -> {
+      this.producer.commitTransaction();
+      future.complete();
+    }, handler);
+    return this;
+  }
+
+  @Override
+  public KafkaWriteStream<K, V> abortTransaction(Handler<AsyncResult<Void>> handler) {
+    this.context.executeBlocking(future -> {
+      this.producer.abortTransaction();
+      future.complete();
+    }, handler);
+    return this;
```

Comment:
These implementations assume that the call on the producer always success and you complete the future.
Should we have a try catch around the producer call to catch exceptions raised by the native producer on the native methods so that we can call the fail on the future with the related cause?

Reply:
Indeed. 👍  I added a better test coverage and assert now for the exception to ensure it is propagated.